src/anim/evaluator/anim-clip.js

```diff
@@ -83,13 +89,52 @@ class AnimClip {
         this._blendOrder = blendOrder;
     }
 
+    get eventCursor() {
+        return this._eventCursor;
+    }
+
+    set eventCursor(value) {
+        this._eventCursor = value;
+    }
+
+    activeEventsForFrame(frameStartTime, frameEndTime) {
+        if (frameStartTime === 0) {
```

Comment:
what if the animation does not start from time 0? This won't be reset to start and won't trigger events.

Reply:
The AnimClip's constructor moves the event cursor to the correct position in the cases when the time is greater than 0.

Reply:
so what's the purpose of this code then? I think you should just do 
this.eventCursor = 0;
at the place you call it with 0, as I assume that's the case it handles.